### ISSUE TITLE
Accessibility Standard Enforcement with eslint-plugin-jsx-a11y Plugin

### ISSUE THREAD

**Issue by SudharakaP** [opencollective/opencollective#3980]
## User story 

I had this idea while reading, https://github.com/opencollective/opencollective-frontend/pull/5896. I think a useful way in the direction of enforcing  accessibility standards is to integrate [`eslint-plugin-jsx-a11y`](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y) to our linter. Maybe we can start with a few basic rules and gradually improve upon on it. Let me know your thoughts. 😄 

Reference: https://reactjs.org/docs/accessibility.html

cc: @Betree

**Comment by Betree** [comment]
I'm up for that! To introduce it smoothly, I would suggest marking all rules generating more than 5 errors as "warning".

The other ones can either be fixed in the same PR, or be marked as warnings too.

**Comment by SudharakaP** [comment]
@Betree : There is a caveat in this approach. The `eslint-plugin-jsx-a11y` doesn't work with `styled-components` very well. So initially when I configured it, it gave around 50 warnings and non on the styled-components which makes me think this is not the best approach. This problem is raised and discussed in https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/466. Now there seems to be another plugin, [eslint-plugin-styled-components-a11y](https://github.com/brendanmorrell/eslint-plugin-styled-components-a11y) which extends `eslint-plugin-jsx-a11y` for styled components. We can integrate that one, but I wanted to double check as that one has 60 stars and probably not very widely used. WDYT? 🤔

**Comment by SudharakaP** [comment]
I've tried to integrate `eslint-plugin-styled-components-a11y` as well but it seems it has a bug so I think at this point we can't use it either. I've create a issue on that repo; https://github.com/brendanmorrell/eslint-plugin-styled-components-a11y/issues/15

**Comment by brendanmorrell** [comment]
@SudharakaP I am just looking at the issue you opened in [the repo](https://github.com/brendanmorrell/eslint-plugin-styled-components-a11y). No one else has reported that error, but I'll pull down your code and see what the issue is in this specific case. I'll let you know when I have a fix up. Thanks for pointing it out!

**Comment by SudharakaP** [comment]
> @SudharakaP I am just looking at the issue you opened in [the repo](https://github.com/brendanmorrell/eslint-plugin-styled-components-a11y). No one else has reported that error, but I'll pull down your code and see what the issue is in this specific case. I approved a PR recently which did not seem like it would cause issues, but maybe it introduced a small bug. I'll let you know when I have a fix up. I'll try and make it quick because, in terms of looking for another solution, I can pretty confidently say none of them exist, which is why I ended up making the package after searching fruitlessly for about a year. Anyway, thanks for pointing the bug out, and I'll update you when it's good to go!

@brendanmorrell : Thanks for the quick response. Yes, we will definitely give this a try once we resolve this issue. Thanks much for looking into it. 😄

**Comment by brendanmorrell** [comment]
@SudharakaP The issue should be fixed now. I just ran eslint on the entire `opencollective/opencollective-frontend` repo with no issues, so if you bump the version to `0.0.32` you should be good to go.

**Comment by SudharakaP** [comment]
> @SudharakaP The issue should be fixed now. I just ran eslint on the entire `opencollective/opencollective-frontend` repo with no issues, so if you bump the version to `0.0.32` you should be good to go.

@brendanmorrell : Thanks so much for the quick fix. I've tested it and seems to work fine. 👍🏽 

@Betree : So I've added `styled-components-a11y` in https://github.com/opencollective/opencollective-frontend/pull/5910. I've made some minor changes to get rid of the most obvious of a11y issues, but for others I've added the eslint rules to just show as warning. Now the majority of the warnings comes due to `href` not being present in our `Link` tags; which will be resolved after merging https://github.com/opencollective/opencollective-frontend/pull/5855. 😄

**Comment by Betree** [comment]
Makes sense! Let's wait for opencollective/opencollective-frontend#5855, I'm confident we can merge it this week
